haskell/test/Test/Ligo/BaseDAO/Proposal/Propose.hs

```diff
@@ -521,3 +522,44 @@ proposalBoundedValue originateFn = do
     call dodDao (Call @"Propose") params
       & expectCustomErrorNoArg #mAX_PROPOSALS_REACHED
 
+proposalStressTest
+  :: forall caps base m.
+    ( MonadNettest caps base m
+    , HasCallStack
+    )
+  => (ConfigDesc Config -> OriginateFn m) -> m ()
+proposalStressTest originateFn = do
+  DaoOriginateData {..} <- originateFn
+    ((ConfigDesc $ configConsts { cmMaxProposals = Just 1000 }) >>-
+     (ConfigDesc $ Period 100) >>- testConfig) defaultQuorumThreshold
+
+  withSender dodOwner1 $
+    call dodDao (Call @"Freeze") (#amount .! 100000)
+
+  originLevel <- getOriginationLevel dodDao
+
+  -- Ensure that we are in a proposal period, or wait, if we are not yet there.
+  let ensureProposalPeriod = do
+        nowLevel <- getLevel
+        let periodNum = div ((nowLevel + 10) - originLevel) dodPeriod
+        when (even periodNum) $ do
+          let waitTill = originLevel + (periodNum + 1)*dodPeriod
+          advanceToLevel waitTill
+
+  -- Create a thousand proposals in batchs of 100
+  forM_ [0..(99 :: Natural)] $ \count -> do
+    let multiplier = 10
```

Comment:
I'm wondering what would be the maximum proposals we can handle. Did you try let say 1 million and see if this causes error or not?